### PR TITLE
Geode 9375 implement zrange radish command

### PR DESCRIPTION
**Pull request by ringles**
GEODE-9375: Implement the basic ZRANGE command to return ranges of entries from sorted sets (zsets).

### CODE REVIEW

**Review by onichols-pivotal — APPROVED** [review]

**Review by dickcav — APPROVED** [review]

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]

**Review by rhoughton-pivot — APPROVED** [review]

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java
Comment (jdeppe-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java:212]: This could have a serious memory impact if the given range is large - compounded since the caller (`zrange`) ends up creating another List. Probably would be better to just inline this method.

Comment (ringles) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java:212]: Hmm. The problem is that here, the OrderStatisticsTree only has OrderedSetEntries. So we're getting a list of OrderedSetEntries out of OrderStatisticsTree, and then in RedisSortedSet.zrange() we're returning a list of byte[]'s. The bytes have to be extracted _somehow_. I'd like to hold off and see how we implement a couple of the other range commands, to see what the best way to refactor this is.

File: geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeIntegrationTest.java
Comment (ringles) [reply geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeIntegrationTest.java]: Good catch - it revealed that subList() didn't work the same way!

File: geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeIntegrationTest.java
Comment (ringles) [reply geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeIntegrationTest.java]: Done, and fixed.

File: geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeIntegrationTest.java
Comment (ringles) [reply geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeIntegrationTest.java:43]: Done.

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java
Comment (jdeppe-pivotal) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java:212]: OK - I just wouldn't want that to get lost. Another option is to return an `Iterator` here. It seems fairly trivial to augment the internal `TreeIterator` with a 'limit'. Anyway, your call.

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/IndexibleTreeSet.java
Comment (jdeppe-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/IndexibleTreeSet.java:54]: The return type should just be a `List` here.

Comment (ringles) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/IndexibleTreeSet.java]: Obviated by changing the return type to Iterator<E>.

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java
Comment (ringles) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java]: An iterator's an excellent idea; I've pushed up an implementation!

Comment (jdeppe-pivotal) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java]: Nice. I like it!

**Review by jdeppe-pivotal — APPROVED** [review]

**Pull request merged**
